dallinger/recruiters.py

```diff
@@ -272,9 +307,27 @@ def get_qualifications(self, hit_id, sandbox):
         """Return the JSON file containing rules to filter participants."""
         raise NotImplementedError
 
-    def get_status(self):
-        """Return the status of the recruiter as a dictionary."""
-        return {}
+    def get_status(self) -> RecruitmentStatus:
+        """Return the status of the recruiter as a RecruitmentStatus."""
+        all_participants = Participant.query.all()
+        statuses = [participant.status for participant in all_participants]
+        status_counts = dict(Counter(statuses))
+        hit_ids = list(set([participant.hit_id for participant in all_participants]))
+        study_id = hit_ids[0] if len(hit_ids) == 1 else ""
+        study_cost = 0
+        for participant in all_participants:
+            base_pay = 0 if participant.base_pay is None else participant.base_pay
+            bonus = 0 if participant.bonus is None else participant.bonus
+            study_cost += base_pay + bonus
+
+        return RecruitmentStatus(
+            recruiter_name=self.nickname,
+            participant_status_counts=status_counts,
+            study_id=study_id,
+            study_status="",
+            study_cost=study_cost,
+            metadata={},
```

Comment:
Seems a bit ugly to have this `metadata` attribute when you could make a `ProlificRecruitmentStatus` subclass where these attributes are properly defined?